src/integrations/prefect-redis/tests/test_messaging.py

```diff
@@ -346,6 +358,65 @@ async def handler(message: Message):
         assert message.attributes == {"id": str(i)}
 
 
+async def test_trimming_streams(
```

Comment:
Great test!

Reply:
Ha, thx!  It took me 10x longer than the actual trimming.  It started rough with just getting the pub/sub working in one test.  Then the test needed to handle a lot of nondeterminism throughout, like how many messages are delivered to each consumer, how many messages get trimmed, etc.